components/camel-aws-sns/src/main/java/org/apache/camel/component/aws/sns/SnsProducer.java

```diff
@@ -104,6 +107,26 @@ private String determineMessageStructure(Exchange exchange) {
                     mav.setDataType("String");
                     mav.withStringValue(value.toString());
                     result.put(entry.getKey(), mav);
+                } else if (value instanceof List) {
+                    List<?> valueList = ((List<?>) value).stream().filter(Objects::nonNull).collect(Collectors.toList());
```

Comment:
With null check I meant to support `null` keyword, not excluding null values. null is according to documentation of `String.Array` valid value. Eg for list `Arrays.asList(null, "element1", "element2", null)` the result should be `[null, "element1", "element2", null]`.

I think this can be replaced with something like this:

```
String result = ((List<?>)values).stream()
    .map(o -> o instanceof String ? String.format("\"%s\"", o) : Objects.toString(o)) // null instanceof String is false and Objects.toString(null) returns String null - this is what we want
    .collect(Collectors.joining(", "));`

MessageAttributeValue mav = new MessageAttributeValue();
mav.setDataType("String.Array");
mav.withStringValue("[" + result + "]");
result.put(entry.getKey(), mav);
```

Reply:
But aws doc says otherwise.

"Name, type, and value must not be empty or null."

https://docs.aws.amazon.com/sns/latest/dg/sns-message-attributes.html

Reply:
But the value will not be empty, the value will be `[null, "element1", "element2", null]`. I am looking into same documentation and it says: 

> String.Array – An array, formatted as a string, that can contain multiple values. The values can be strings, numbers, or the keywords true, false, and **null**.

Reply:
Oh, I missed that. Will make null a valid value then. 
Also was looking at your suggestion, "o instanceof String ? String.format("\"%s\"", o)". Don't you think it's unnecessary to do instanceof check on every value ?